### PR TITLE
OPIK-501 Basic experiment items CRUD tests

### PR DESCRIPTION
**Pull request by AndreiCautisanu**
## Details

Tests proper creation and deletion of experiment items, on both UI and backend.
Coverage:

1. Checks that all experiment items created as part of an experiment creation are present on both the UI and the backend and checks that all the IDs match
2. Checks that after deletion of an experiment item, both the backend and the UI are properly updated (count decreased, list of IDs still matching (same items fetched from both)).

**Notes**:
- Currently using FERN generated code directly to access the backend as there is no functionality to grab experiment/experiment item info via the SDK, will adapt the tests if and when such functionality is added.
- Currently only testing creation and deletion of the items themselves, more specific tests regarding the specific content of the items (corresponding dataset item content, correct metric values) to be revisited later after I get all CRUD coverage and first improvements (better reports, running against cloud envs) done

### CODE REVIEW

**Comment by AndreiCautisanu** [comment]
Tests are passing https://github.com/comet-ml/opik/actions/runs/12278915929/job/34261702734

File: tests_end_to_end/page_objects/ExperimentItemsPage.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def get_all_item_ids_in_experiment(self):
+        ids = []
+        ids.extend(self.get_all_item_ids_on_current_page())
+        while self.next_page_button_locator.is_visible() and self.next_page_button_locator.is_enabled():
+            self.next_page_button_locator.click()
+            self.page.wait_for_timeout(500)
+            ids.extend(self.get_all_dataset_items_on_current_page())
+
+        return ids
```
Comment (andrescrz) [inline tests_end_to_end/page_objects/ExperimentItemsPage.py:53]: Minor: mentioned before, but please add a pre-commit hook to format the Python code within this `tests_end_to_end` folder, so among other things, it adds an empty line at the end of each file. Use the SDK linting as reference.

File: tests_end_to_end/tests/Datasets/datasets_utils.py [hunk 2]
```diff
@@ -48,6 +48,7 @@ def insert_dataset_items_ui(page: Page, dataset_name, items_list):
     
     for item in items_list:
         dataset_items_page.insert_dataset_item(json.dumps(item))
+        time.sleep(0.2)
```
Comment (andrescrz) [inline tests_end_to_end/tests/Datasets/datasets_utils.py:51]: Any particular reason to throttle the insertion pace here? The recommended alternative is inserting dataset items in batches, so you can insert the whole `items_list` in one go.

**Review by andrescrz — DISMISSED** [review]
Couple of minor comments for potential follow-ups, but not blockers. Everything else LGTM.

File: tests_end_to_end/page_objects/ExperimentItemsPage.py (hunk 1 above)
Comment (AndreiCautisanu) [reply tests_end_to_end/page_objects/ExperimentItemsPage.py:53]: good point, i went ahead and added full linting functionality, along with a workflow file to check on every PR. applied it as part of this PR, everything looks good. will add more bits in the future when doing improvements (i.e. clean typing on everything like the SDK code currently has)

File: tests_end_to_end/tests/Datasets/datasets_utils.py (hunk 2 above)
Comment (AndreiCautisanu) [reply tests_end_to_end/tests/Datasets/datasets_utils.py:51]: The main bit of flakiness that's currently present in these tests is one dataset item you insert via the UI not registering if you switch away too quickly. Added a small delay to try and prevent that and see if it works. As far as I know there is no way to add a list of dataset items via the UI, only one at a time, which is why I do it this way here. When inserting via the SDK i just pass the entire list in one go

**Review by andrescrz — CHANGES_REQUESTED** [review]
The linting is great, but apologies because I confused you. I meant to send it in a separated PR, otherwise is hard to review. Let's separate this into:
1. Basic experiment items CRUD tests
2. All the linting infra and the updated files.

**Comment by AndreiCautisanu** [comment]
@andrescrz all good now, reverted the linting fixes and will get them in via a different PR

**Review by andrescrz — APPROVED** [review]

**Pull request merged**